src/directives/AddrResolver.js

```diff
@@ -34,157 +30,65 @@ const AddrResolver = {
-                    errorPar.innerText = _this.$t('ens.ens-resolver.network-not-found', { network: network.type.name[0]});
-                      _this.isValidAddress = false;
-                      _this.hexAddress = '';
-                      vnode.elm.parentNode.parentNode.appendChild(errorPar);
-                    });
-                } else {
-                  removeElements();
-                  // eslint-disable-next-line
-                    errorPar.innerText = _this.$t('ens.ens-resolver.network-not-found', { network: network.type.name[0]});
-                  _this.isValidAddress = false;
-                  _this.hexAddress = '';
-                  vnode.elm.parentNode.parentNode.appendChild(errorPar);
-                }
-              });
-          }
-        }
-      } else if (e !== '') {
-        const isValid = WAValidator.validate(e, parentCurrency);
-        if (isValid) {
-          _this.isValidAddress = isValid;
-          _this.hexAddress =
-            parentCurrency === 'ETH' ? toChecksumAddress(e) : e;
-        } else {
-          if (canValidate(parentCurrency)) {
-            const isValid = MAValidator.validate(e, parentCurrency);
-            if (isValid) {
-              _this.isValidAddress = isValid;
-              _this.hexAddress = e;
-            } else {
-              _this.isValidAddress = false;
-              _this.hexAddress = '';
+      try {
+        const address = await resolution.addressOrThrow(domain, parentCurrency);
+        if (address) {
+          if (!checkDarklist(address)) {
+            const addCheck = WAValidator.validate(e, parentCurrency);
+            if (addCheck) {
+              _this.isValidAddress = addCheck;
+              _this.hexAddress = toChecksumAddress(e);
               removeElements();
-
-              if (e.length > 0) {
-                if (
-                  parentCurrency === 'ETH' &&
-                  (e.length !== 42 || !utils.isHexStrict(e))
-                ) {
-                  errorPar.innerText = _this.$t(
-                    'ens.ens-resolver.invalid-eth-addr'
-                  );
-                } else if (
-                  parentCurrency === 'ETH' &&
-                  !utils.checkAddressChecksum(e)
-                ) {
-                  errorPar.innerText = _this.$t(
-                    'ens.ens-resolver.addr-not-checksummed'
-                  );
-                  // 'Incorrect checksum: check address format on EthVM';
-                } else {
-                  errorPar.innerText = _this.$t(
-                    'ens.ens-resolver.invalid-addr',
-                    { coin: parentCurrency }
-                  );
-                }
-              } else {
-                errorPar.innerText = '';
-              }
-
-              el.parentNode.parentNode.appendChild(errorPar);
+              messagePar.classList.add('resolver-addr');
+              _this.isValidAddress = true;
+              _this.hexAddress = toChecksumAddress(address);
```

Comment:
`_this.isValidAddress = true;` and `_this.hexAddress = toChecksumAddress(address);` conflicts with the same assignment on line 66 and 67. These should be an else on this if statement or the other way around.